electrolytes/data2lammps.py

```diff
@@ -0,0 +1,326 @@
+    'Er': 167.26,
+    'Tm': 168.93,
+    'Yb': 173.05,
+    'Lu': 174.97,
+    'Hf': 178.49,
+    'Ta': 180.95,
+    'W': 183.84,
+    'Re': 186.21,
+    'Os': 190.23,
+    'Ir': 192.22,
+    'Pt': 195.08,
+    'Au': 196.97,
+    'Hg': 200.59,
+    'Tl': 204.38,
+    'Pb': 207.2,
+    'Bi': 208.98,
+    'Po': 209.0,
+    'At': 210.0,
+    'Rn': 222.0,
+    'Fr': 223.0,
+    'Ra': 226.0,
+    'Ac': 227.0,
+    'Th': 232.04,
+    'Pa': 231.04,
+    'U': 238.03,
+    'Np': 237.0,
+    'Pu': 244.0,
+    'Am': 243.0,
+    'Cm': 247.0,
+    'Bk': 247.0,
+    'Cf': 251.0,
+    'Es': 252.0,
+    'Fm': 257.0,
+    'Md': 258.0,
+    'No': 259.0,
+    'Lr': 266.0,
+    'Rf': 267.0,
+    'Db': 270.0,
+    'Sg': 271.0,
+    'Bh': 270.0,
+    'Hs': 277.0,
+    'Mt': 276.0,
+}
+
+
+def get_indices(comments, keyword):
+    """ Grab indices of labeled columns given a specific keyword. 
+        Args: 
+            comments (list): a list of strings, each of which is a label for a specific column
+            keyword (string): specific keyword we want to match with comments.
+
+        This helper function is specific to reading the elytes.csv file.
+    """
+    indices = [i for i, string in enumerate(comments) if keyword in string]
+    return indices[::2], indices[1::2]
+    
+def get_species_and_conc(systems, i, indices):
+    """ Grab list of species and their corresponding concentrations 
+        Args: 
+            systems (list): a list of lists, containing the full spreadhseet information from the CSV file. 
+            i (int): index of the row of the spreadsheet we want to access. Numbering starts from one. 
+            indices (list): index of columns that we want to access. 
+
+        This helper function is specific to reading the elytes.xls file.
+    """
+    species = np.array(systems[i])[indices]
+    return [name for name in species if name]
+
+def run_packmol_moltemplate(species,boxsize,Nmols,filename,directory):
+    """ Run Packmol and Moltemplate to generate system configuration (in LAMMPS data format) 
+        as well as files to run a LAMMPS simulation.
+
+        Args:
+            species (list): list of molecular species name (string) that we want to simulate
+            boxsize (float): size of the simulation box, using Angstrom units
+            Nmols (list): list of number of molecules we want to pack inside the simulation box
+            filename (string): a prefix name for all files we want to generate, including LAMMPS DATA files and PDB files.
+            directory (string): directory to generate all the output files
+    """
+    try:
+        os.mkdir(directory)
+    except Exception as e:
+        print(e)
+    # Copy-paste the LT file, which has the OPLS settings
+    general_ff = "oplscm1a.lt"
+    bashCommand = f'cp {general_ff} ./{directory}'
+    os.system(bashCommand)
+   
+
+    # Prepare the Packmol script
+    packmolstring =f"""
+    tolerance 2.0
+    filetype pdb
+    output {filename}.pdb
+    """
```

Comment:
You have to install packmol and being able to run it in bash for it to work. 

Looking at their source code (https://mdgo.readthedocs.io/en/latest/_modules/mdgo/mdgopackmol.html#PackmolWrapper) it works the same way. They use subprocess.run instead of os.system that I do. Would you say subprocess is a cleaner way?

I wanna stick with my code because I try to minimize the number of dependent packages I use. 

Also: Hmm, maybe I should make it a user parameter and default it to 2.0

Reply:
subprocess is definitely cleaner, I made the same comment below.